lib/IDE/SyntaxModel.cpp

```diff
@@ -766,6 +766,10 @@ bool ModelASTWalker::walkToDeclPre(Decl *D) {
                                                      AFD->getBodySourceRange());
       SN.NameRange = charSourceRangeFromSourceRange(SM,
                           AFD->getSignatureSourceRange());
+      if (FD) {
+        SN.TypeRange = charSourceRangeFromSourceRange(SM,
+                                      FD->getReturnTypeLoc().getSourceRange());
```

Comment:
I did that originally, but `walkToSubStructurePre` in `SwiftEditor.cpp` already checks if `TypeRange` is valid. Should I add a check here too?